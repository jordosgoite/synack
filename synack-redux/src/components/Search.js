import React from 'react';
import { useState } from 'react';
import { useDispatch } from 'react-redux';
import Modal from './parts/Modal';
import fetchData from './redux/actions/SearchActions';


const Search =()=>{
    const dispatch =  useDispatch();
    const [modal, setModal]=useState(false);
    const [mySearch, setMySearch] = useState({
        searchData: '',
        searchEngine:''
    });
    const handleChange = (e) =>{
        setMySearch({...mySearch, [e.target.name]:e.target.value})
    }
    const handleModal = (modal) =>{
        setModal(modal)
    }
    const validate = () =>{
        let isValid = true;
        if(!mySearch.searchData){
            isValid = false;
        }
        if(!mySearch.searchEngine){
            isValid = false;
        }
        return isValid
    }
    const handleSubmit = () =>{
        if (validate()){
            dispatch(fetchData(mySearch))
        }
        else{
            setModal(true)
        }
    }
    return(
        <div className="search-menu">
            <input className="search-input" type="text" name="searchData" placeholder="Search..."
                onChange={handleChange}
            />
            <select className="search-select" name="searchEngine" onChange={handleChange}>
                <option defaultValue="">Choose Search Engine</option>
                <option value='google'>Google</option>
                <option value='bing'>Bing</option>
                <option value='both'>Both</option>
            </select>
            <button className="search-button" onClick={handleSubmit}>Search</button>
            {modal ? <Modal modal={handleModal} /> : null}
        </div>
    );
}
export default Search