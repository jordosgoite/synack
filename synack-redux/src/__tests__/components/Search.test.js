import React from 'react'
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'
import Search from '../../components/Search.js'
import fetchData from '../../components/redux/actions/SearchActions'

const mockStore = configureStore([]);

describe('React-Redux Search Component', () => {
    let store;
    let component;
   
    beforeEach(() => {
      store = mockStore({
        myState: 'sample text',
      });
   
      component = renderer.create(
        <Provider store={store}>
          <Search />
        </Provider>
      );
    });
   
    it('should render with given state from Redux store', () => {
      expect(component.toJSON()).toMatchSnapshot();
    });
    it('should render a button', () => {
        renderer.act(() => {
          component.root.findByType('button').props.onClick();
        });
      });
  });

