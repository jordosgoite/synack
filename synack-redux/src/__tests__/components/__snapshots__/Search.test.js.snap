// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`React-Redux Search Component should render with given state from Redux store 1`] = `
<div
  className="search-menu"
>
  <input
    className="search-input"
    name="searchData"
    onChange={[Function]}
    placeholder="Search..."
    type="text"
  />
  <select
    className="search-select"
    name="searchEngine"
    onChange={[Function]}
  >
    <option
      defaultValue=""
    >
      Choose Search Engine
    </option>
    <option
      value="google"
    >
      Google
    </option>
    <option
      value="bing"
    >
      Bing
    </option>
    <option
      value="both"
    >
      Both
    </option>
  </select>
  <button
    className="search-button"
    onClick={[Function]}
  >
    Search
  </button>
</div>
`;
